src/engine/enginemaster.cpp

```diff
@@ -428,12 +428,14 @@ void EngineMaster::process(const int iBufferSize) {
         // Process effects and mix PFL channels together for the headphones.
         // Effects will be reprocessed post-fader for the crossfader buses
         // and master mix, so the channel input buffers cannot be modified here.
-        ChannelMixer::applyEffectsAndMixChannels(
-            m_headphoneGain, &m_activeHeadphoneChannels,
-            &m_channelHeadphoneGainCache,
-            m_pHead, m_headphoneHandle.handle(),
-            m_iBufferSize, m_iSampleRate,
-            m_pEngineEffectsManager);
+        ChannelMixer::applyEffectsAndMixChannels(m_headphoneGain,
+                &m_activeHeadphoneChannels,
+                &m_channelHeadphoneGainCache,
+                m_pHead,
+                m_headphoneHandle.handle(),
+                m_iBufferSize,
+                static_cast<int>(m_sampleRate.value()),
```

Comment:
Is the explicit cast from unsigned to signed needed?

Reply:
SampleRate uses `uint32_t` internally. I think `int` is guaranteed to have 4 bytes, but has a different value range because it's signed. I didn't check if removing the cast results in a conversion warning but if it doesn't it should.

I didn't add a `toInt` method because in the end we should just use the SampleRate type in ChannelMixer, too. I just left out these changes in favor of a smaller diff and less conflicts with #2618 and #1966.